File: FangZhiHu/app/src/main/java/network/ProgressDownSubscriber.java
package network;



import io.reactivex.subscribers.ResourceSubscriber;
import network.download.DownInfo;

public class ProgressDownSubscriber<T> extends ResourceSubscriber<T> {
    /*下载数据*/
    private DownInfo downInfo;

    public ProgressDownSubscriber() {
        super();
    }

    public ProgressDownSubscriber(DownInfo downInfo) {
        this.downInfo=downInfo;
    }

    @Override
    public void onStart() {
        super.onStart();
        if(downInfo.getmListener() != null){
            downInfo.getmListener().onStart();
        }
        downInfo.state = DownInfo.DownState.START;
    }

    @Override
    public void onError(Throwable e) {
        if(downInfo.getmListener() != null){
            downInfo.getmListener().onError(e);
        }
        downInfo.state = DownInfo.DownState.ERROR;
    }

    @Override
    public void onComplete() {
        if(downInfo.getmListener() != null){
            downInfo.getmListener().onComplete();
        }
        downInfo.state = DownInfo.DownState.COMPLETE;
    }

    @Override
    public void onNext(T t) {
    }

}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/ProgressDownSubscriber.kt
package network

import io.reactivex.subscribers.ResourceSubscriber
import network.download.DownInfo
import network.download.DownInfo.DownState.*
import network.download.DownProgressListener

/**
 * Created by tongqinyuan on 2018/3/8.
 */
class ProgressDownSubscriber<T> :ResourceSubscriber<T> {
    var mDownInfo:DownInfo? = null
    constructor()
    constructor(downInfo: DownInfo){
        mDownInfo = downInfo
    }

    override fun onStart() {
        super.onStart()
        if(mDownInfo != null){
            mDownInfo!!.getListener()!!.onStart()
        }
        mDownInfo!!.state = START
    }



    override fun onNext(t: T) {

    }

    override fun onError(t: Throwable?) {
        if(mDownInfo != null){
            mDownInfo!!.getListener()!!.onError()
        }
        mDownInfo!!.state = ERROR
    }

    override fun onComplete() {
        if(mDownInfo != null){
            mDownInfo!!.getListener()!!.onComplete()
        }
        mDownInfo!!.state = COMPLETE
    }
}